### ISSUE TITLE
Deserialization fails with `XmlMapper` and `DeserializationFeature.UNWRAP_ROOT_VALUE`

### ISSUE THREAD

**Issue by dewarim** [FasterXML/jackson-dataformat-xml#374]
Simple Test case:  https://github.com/dewarim/jackson-xml-debug

Using jackson-dataformat-xml version 2.10.1 @ Java 11

Given: a POJO:
```
@JacksonXmlRootElement(localName = "Root")
@JsonRootName("Root")
public class Root {

    @JsonProperty("id")
    String id;

    public Root(String id) {
        this.id = id;
    }

    public Root() {
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (o == null || getClass() != o.getClass()) {
            return false;
        }
        Root root = (Root) o;
        return Objects.equals(id, root.id);
    }

    @Override
    public int hashCode() {
        return Objects.hash(id);
    }
}
```

When trying to do 
```
        ObjectMapper mapper = new XmlMapper();
       mapper.enable(DeserializationFeature.UNWRAP_ROOT_VALUE);
        Root deserializedRoot = mapper.readValue(value, Root.class);
```

I get:
com.fasterxml.jackson.databind.exc.MismatchedInputException: Root name 'id' does not match expected ('Root') for type [simple type, class test.debug.Root]
 at [Source: (StringReader); line: 1, column: 7] (through reference chain: test.debug.Root["id"])

**Comment by cowtowncoder** [comment]
For sake of completeness (and since example above is missing a key piece, XML document), test itself is:

```
    @Test
    public void xml() throws JsonProcessingException {
        ObjectMapper mapper = new XmlMapper();
        mapper.enable(SerializationFeature.WRAP_ROOT_VALUE);
        mapper.enable(DeserializationFeature.UNWRAP_ROOT_VALUE);
        String value = mapper.writeValueAsString(root);
        assertEquals(<Root><id>hello</id></Root>", value);
        Root deserializedRoot = mapper.readValue(value, Root.class);
        assertEquals(root, deserializedRoot);
    }
```

**Comment by cowtowncoder** [comment]
I think I am actually inclined to work on changing things in 2.11 so that both `WRAP_ROOT_VALUE` and `UNWRAP_ROOT_VALUE` are ignored for XML  backend: currently only former is.

**Comment by cowtowncoder** [comment]
Where there's will there's a way. So returning to this issue now; fix imminent.

**Comment by ionel-sirbu-crunch** [comment]
Hi, what is the fix after all? Can we still achieve root element unwrapping? If so, how?
I've just switched to v. 2.12.3 and unit tests started failing because this no longer works.

**Comment by cowtowncoder** [comment]
@ionel-sirbu-crunch Unwrapping in XML has different semantics than with JSON so that feature does nothing with XML. Fix is to prevent attempts for unwrapping since that will not work the way databind would try to apply it.

For your specific problem, please file a new issue with reproduction (showing what specifically you think is failing) and include information on versions where things appeared to work.

**Comment by ionel-sirbu-crunch** [comment]
> @ionel-sirbu-crunch Unwrapping in XML has different semantics than with JSON so that feature does nothing with XML. Fix is to prevent attempts for unwrapping since that will not work the way databind would try to apply it.
> 
> For your specific problem, please file a new issue with reproduction (showing what specifically you think is failing) and include information on versions where things appeared to work.

Created https://github.com/FasterXML/jackson-dataformat-xml/issues/485 for that matter. Thank you!

**Comment by cowtowncoder** [comment]
Note: as per #485 the behavior WILL CHANGE for 2.13.0 so that:

* Deserializer WILL AGAIN unwrap root element, but
* Serializer STILL DOES NOT add wrapping

Second issue may be addressed with a later version (2.14 or later) but first we will want to get back to 2.11 baseline compatibility.
